integration-cli/docker_cli_volume_test.go

```diff
@@ -431,6 +431,30 @@ func (s *DockerSuite) TestVolumeCLIRmForce(c *check.C) {
 	c.Assert(out, checker.Contains, name)
 }
 
+// TestVolumeCLIRmForceInUse verifies that repeated `docker volume rm -f` calls does not remove a volume
+// if it is in use. Test case for https://github.com/docker/docker/issues/31446
+func (s *DockerSuite) TestVolumeCLIRmForceInUse(c *check.C) {
+	name := "testvolume"
+	out, _ := dockerCmd(c, "volume", "create", name)
+	id := strings.TrimSpace(out)
+	c.Assert(id, checker.Equals, name)
+
+	prefix, slash := getPrefixAndSlashFromDaemonPlatform()
+	dockerCmd(c, "create", "-v", "testvolume:"+prefix+slash+"foo", "busybox")
+
+	_, _, err := dockerCmdWithError("volume", "rm", "-f", name)
+	c.Assert(err, check.NotNil)
+	c.Assert(err.Error(), checker.Contains, "volume is in use")
+	out, _ = dockerCmd(c, "volume", "ls")
+	c.Assert(out, checker.Contains, name)
+
+	_, _, err = dockerCmdWithError("volume", "rm", "-f", name)
```

Comment:
Checking it twice, because that's how I reproduced the original issue; first `rm -f` "passed", but the volume was still present in `volume ls`. The second `rm -f` actually removed the volume (also from the list).

Was in doubt if I should add the final remove. Small change, so I can add that yes